native_client/deepspeech.h

```diff
@@ -109,6 +112,25 @@ char* DS_SpeechToText(ModelState* aCtx,
                       unsigned int aBufferSize,
                       unsigned int aSampleRate);
 
+/**
+ * @brief Use the DeepSpeech model to perform Speech-To-Text and output metadata 
+ * about the results.
+ *
+ * @param aCtx The ModelState pointer for the model to use.
+ * @param aBuffer A 16-bit, mono raw audio signal at the appropriate
+ *                sample rate.
+ * @param aBufferSize The number of samples in the audio signal.
+ * @param aSampleRate The sample-rate of the audio signal.
+ *
+ * @return Outputs a struct of individual letters along with their timing information. 
+ * 		   The user is responsible for freeing Metadata and Metadata.items. Returns NULL on error.
```

Comment:
`MetadataItem::character` also needs to be freed, right? Maybe we should provide a helper function.

```c++
DEEPSPEECH_EXPORT
void DS_FreeMetadata(Metadata* m) {
    for (int i = 0; i < m->num_items; ++i) {
        free(m->items[i]->character);
    }
    free(m->items);
    free(m);
}
```

Reply:
The characters don't actually need to be freed - it crashed when I did this. I still added the function though.